pageserver/src/tenant/timeline.rs

```diff
@@ -3954,6 +3896,9 @@ impl Timeline {
             //
             // This does not in fact have any effect as we no longer consider local metadata unless
             // running without remote storage.
+            //
+            // This unconditionally schedules also an index_part.json update, even though, we will
+            // be doing one a bit later with the unlinked gc'd layers.
             self.update_metadata_file(self.disk_consistent_lsn.load(), None)
                 .await?;
```

Comment:
I think this was required by why test_gc_cutoff was introduced in #2558 which was introduced to fix #2539. Also I think #5624 should just be closed as duplicate.

Reading Kirills comment and this comment of yours I think this can now be removed: https://github.com/neondatabase/neon/issues/2539#issuecomment-1265801283

Reply:
But yeah, as we still don't require #4099 cannot remove this. (Even if the non-remote_storage one is currently broken.)

Reply:
`doc: cleanup, link 4099` perhaps fbba48830